### ISSUE TITLE
ebay-select optgroup without children

### ISSUE THREAD

**Issue by venkganesan** [eBay/ebayui-core#1110]
# Feature Request

## Description

In ebay-select there is no way to just add an optgroup with no option children. https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-select/index.marko#L49.

## Context

I am trying to recreate ebay's default search category dropdown which uses an empty optgroup to show a divider. Using the current ebay-select this is not possible as an empty option always gets added. Please add a way to create an optgroup without children.

## Screenshots
<img width="371" alt="Screen Shot 2020-05-01 at 2 48 21 PM" src="https://user-images.githubusercontent.com/64790772/80975745-7e5e5400-8df0-11ea-9d77-c8868348b6fa.png">

**Comment by ianmcburnie** [comment]
After more consideration and investigation, I'm not convinced this is a great idea to use an empty `optgroup` as a purely presentational element. I tried it with VoiceOver in Safari and it is unable to announce the label, possibly due to this formatting, and therefore would not be accessible to those users. For more fine-grained visual control we recommend using `<ebay-listbox-button>` for desktop.

**Comment by agliga** [comment]
So doing research for this, I think the best option is to do a disabled option with text as `-------`

So using the current `ebay-select` you can add: `<ebay-select-option disabled text="------------------" />`
This works the best and is the best crossbrowser solution. There is no change need on our end to support this since the markup presented here is simple enough

Here is a stackoverflow discussion about adding separators: https://stackoverflow.com/questions/899148/html-select-option-separator

**Comment by agliga** [comment]
<img width="713" alt="Screen Shot 2020-06-22 at 10 31 27 AM" src="https://user-images.githubusercontent.com/1755269/85317818-e1c14580-b473-11ea-8560-dc06650c03d7.png">
Here is a mockup of an example I was playing with

**Comment by ianmcburnie** [comment]
Looks good.

**Comment by ianmcburnie** [comment]
@venkganesan does this work for you?

**Comment by venkganesan** [comment]
Yes, this works. Thank you 👍
